### ISSUE TITLE
hostapd causes kernel panic

### ISSUE THREAD

**Issue by jschultz** [Mange/rtl8192eu-linux-driver#273]
Trying to use this driver with hostapd with a DLink TL-WN822N AP on a Raspberry Pi B (recent image, kernel version 5.15.32) I found that it worked very briefly before disconnecting. The logs showed a kernel panic at line 8084 of `core/rtw_mlme_ext.c` which the source shows to be an error trap relating to a too long beacon frame size.

So I put in a bunch of logs and found that the beacon frame size remained constant between calls to `issue_beacon` until a client connected to the AP. At that point the value would increase by 6 from one call to the next, very quickly reaching the maximum size and triggering the panic. I tried more and more logs trying to track where that error might have come from, and could still go down that path though it starts to get complicated. At this stage I'm hoping that someone who actually knows something about Wifi drivers might already recognise what is probably happening.

In particular I note a commented log call at line 8104 of `core/rtw_mlme_ext.c`:

`   /* RTW_INFO("issue bcn_sz=%d\n", pattrib->last_txcmdsz); */`

which seems to be tracking the problem, in that `pattrib->last_txcmdsz` follows the beacon size.

For comparison, I also built the driver on an Intel Debian box (kernel version 5.16.0-3)  and got exactly the same result.

**Comment by CGarces** [comment]
I don't have knowledge to fix this issue.
Your best option is try to use the mainline kernel (RTL8XXXU).
Another option is try to understand similar drivers from linux kernel (specially drivers/staging/rtl*) like this one
https://github.com/torvalds/linux/blob/master/drivers/staging/rtl8723bs/core/rtw_mlme_ext.c
And identify is there is some relevant changes related to `issue_beacon` that should be ported to this driver.

**Comment by jschultz** [comment]
Thanks @CGarces I tried rtl8xxxu but it doesn't support Master mode. I looked at the rtl8723s driver and it seems very similar (but not identical) to rtl8192eu. Perhaps if I had a unit that used that driver I could compare results between the two.

But I found that using Network Manager instead of hostapd to run the AP worked fine. So perhaps the bug is actually in hostapd? Anyway I'm happy with my work-around so just leaving this issue there.
